openmrs-19/src/main/java/org/motechproject/openmrs19/resource/impl/PatientResourceImpl.java

```diff
@@ -121,22 +123,34 @@ public void deletePatient(Config config, String uuid) {
 
     @Override
     public void updatePatientMotechId(Config config, String patientUuid, String newMotechId) {
-        Identifier patientIdentifier = getPatientIdentifier(config, patientUuid);
-        patientIdentifier.setIdentifier(newMotechId);
+        List<Identifier> patientIdentifiers = getPatientIdentifierList(config, patientUuid);
+        for (Identifier patientIdentifier : patientIdentifiers) {
+            if ("MOTECH Id".equals(patientIdentifier.getIdentifierType().getDisplay())) {
+                patientIdentifier.setIdentifier(newMotechId);
+
+                String requestJson = buildGson().toJson(patientIdentifier);
+                postForJson(config, requestJson, "/patient/{patientUuid}/identifier/{identifierUuid}", patientUuid,
+                        patientIdentifier.getUuid());
+                break;
+            }
+        }
+    }
 
-        String requestJson = buildGson().toJson(patientIdentifier);
+    @Override
+    public void updatePatientIdentifier(Config config, String patientUuid, Identifier updatedIdentifier) {
+        String requestJson = buildGson().toJson(updatedIdentifier);
         postForJson(config, requestJson, "/patient/{patientUuid}/identifier/{identifierUuid}", patientUuid,
-                patientIdentifier.getUuid());
+                updatedIdentifier.getUuid());
     }
 
     private PatientIdentifierListResult getAllPatientIdentifierTypes(Config config) {
         String responseJson = getJson(config, "/patientidentifiertype?v=full");
         return (PatientIdentifierListResult) JsonUtils.readJson(responseJson, PatientIdentifierListResult.class);
     }
 
-    private Identifier getPatientIdentifier(Config config, String patientUuid) {
+    public List<Identifier> getPatientIdentifierList(Config config, String patientUuid) {
```

Comment:
additionally, this method should be higher, according to the sequence from service